usr/src/uts/common/os/aio.c

```diff
@@ -1735,21 +1689,17 @@ alio_cleanup(aio_t *aiop, aiocb_t **cbp, int nent, int run_mode)
 	int i;
 	aio_req_t *reqp;
 	aio_result_t *resultp;
-	aiocb64_32_t *aiocb_64;
 
 	for (i = 0; i < nent; i++) {
 		if (get_udatamodel() == DATAMODEL_NATIVE) {
 			if (cbp[i] == NULL)
 				continue;
-			if (run_mode == AIO_LARGEFILE) {
```

Comment:
Just to confirm my understanding of this and similar changes: this happens because we're in the native environment which by definition right now can only by LP64 and therefore that's why we no longer consult the run mode?

Reply:
This is because we're checking `DATAMODEL_NATIVE` which used to be able to mean "32bit process on 32bit kernel", and thus require largefile support.  Now we have no 32bit kernel, a `DATAMODEL_NATIVE` process can never require largefile.